Source/DafnyCore/Verifier/BoogieGenerator.Iterators.cs

```diff
@@ -81,7 +81,7 @@ public partial class BoogieGenerator {
       // FREE PRECONDITIONS
       if (kind == MethodTranslationKind.SpecWellformedness || kind == MethodTranslationKind.Implementation) {  // the other cases have no need for a free precondition
         // free requires mh == ModuleContextHeight && fh = FunctionContextHeight;
-        req.Add(Requires(iter.tok, true, etran.HeightContext(iter), null, null, null));
+        req.Add(Requires(iter.tok, true, null, etran.HeightContext(iter), null, null, null));
```

Comment:
At first I was confused by the `null` parameter here and then realized that these are free requires clauses so they'll never be part of any proof obligation.